### DISCUSSION TITLE
Pre-Commit / Post Commit Hooks

### DISCUSSION THREAD

**Discussion by safaci2000** [go-jet/jet#358] [General]
I have a pattern where I'd like to say always insert a record into a database table when a transaction is active.  I was wondering if there was any thing similar Bob's Hook patterns.  

Example: https://bob.stephenafamo.com/docs/models/hooks

It seems like a very useful features and was curious if Jet had something similar.

**Comment by houtn11** [comment]
You should be able to write it yourself. Is this what you had in mind?

```golang
func Query(stmt Statement, db qrm.Queryable, dest any) error {
	// pre hook
	// .....
	err := stmt.Query(db, &dest)

	// post hook
	// .....

	return err
}
```

**Reply by safaci2000** [reply]
I think the idea of a hook is just having something that gets automatically invoked after every SQL operation or at least after every write on a given table for example.  What you're showing me is two SQL statements unless I'm misreading that?

**Reply by houtn11** [reply]
> What you're showing me is two SQL statements unless I'm misreading that?

From what I can see that's exactly how bob hooks works.
